### PR TITLE
feat: DynamoDB authorization support

### PR DESCRIPTION
**Pull request by janlesage**

### CODE REVIEW

**Review by rikdepeuter — APPROVED** [review]

File: src/RoadRegistry.BackOffice.Abstractions/Configuration/DynamoDbClientOptions.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace RoadRegistry.BackOffice.Abstractions.Configuration;
+
+using Amazon;
+
+public class DynamoDbClientOptions
+{
+    public const string ConfigurationKey = "DynamoDbClientOptions";
+    public string AccessKeyId { get; set; }
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Abstractions/Configuration/DynamoDbClientOptions.cs:8]: AccessKeyId & AccessKeySecret won't be necessary

File: src/RoadRegistry.BackOffice.Api/Changes/ChangeFeedController.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ namespace RoadRegistry.BackOffice.Api.Changes;
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("changefeed")]
 [ApiExplorerSettings(GroupName = "ChangeFeed")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Changes/ChangeFeedController.cs:26]: should be kept to ApiKeyAuth("Road")

File: src/RoadRegistry.BackOffice.Api/Downloads/DownloadController.cs [hunk 3]
```diff
@@ -16,7 +16,7 @@ namespace RoadRegistry.BackOffice.Api.Downloads;
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("download")]
 [ApiExplorerSettings(GroupName = "Downloads")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Downloads/DownloadController.cs:19]: same as above

File: src/RoadRegistry.BackOffice.Api/Extracts/ExtractsController.cs [hunk 4]
```diff
@@ -24,7 +24,7 @@ namespace RoadRegistry.BackOffice.Api.Extracts;
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("extracts")]
 [ApiExplorerSettings(GroupName = "Extracts")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Extracts/ExtractsController.cs:27]: same as above

File: src/RoadRegistry.BackOffice.Api/Information/InformationController.cs [hunk 5]
```diff
@@ -11,14 +9,14 @@ namespace RoadRegistry.BackOffice.Api.Information;
...
-using Newtonsoft.Json;
+using System.Threading;
+using System.Threading.Tasks;
 
 [ApiVersion(Version.Current)]
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("information")]
 [ApiExplorerSettings(GroupName = "Information")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Information/InformationController.cs:19]: same as above

File: src/RoadRegistry.BackOffice.Api/RoadRegistrySystem/RoadRegistrySystemController.cs [hunk 6]
```diff
@@ -13,7 +13,7 @@ namespace RoadRegistry.BackOffice.Api.RoadRegistrySystem;
 [ApiVersion(Version.Current)]
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("system")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/RoadRegistrySystem/RoadRegistrySystemController.cs:16]: should be kept to ApiKeyAuth("Road")

File: src/RoadRegistry.BackOffice.Api/Uploads/UploadController.cs [hunk 7]
```diff
@@ -22,7 +22,7 @@ namespace RoadRegistry.BackOffice.Api.Uploads;
 [AdvertiseApiVersions(Version.CurrentAdvertised)]
 [ApiRoute("upload")]
 [ApiExplorerSettings(GroupName = "Uploads")]
-[ApiKeyAuth("Road")]
+[ApiKeyAuth]
```
Comment (lvermeulen) [inline src/RoadRegistry.BackOffice.Api/Uploads/UploadController.cs:25]: same as above

File: src/RoadRegistry.BackOffice.Abstractions/Configuration/DynamoDbClientOptions.cs (hunk 1 above)
Comment (janlesage) [reply src/RoadRegistry.BackOffice.Abstractions/Configuration/DynamoDbClientOptions.cs:8]: > AccessKeyId & AccessKeySecret won't be necessary

This is required for the DynamoDbClient, same as distributed lock. Identical configuration also can be found inside the application definitions

Comment (lvermeulen) [reply src/RoadRegistry.BackOffice.Abstractions/Configuration/DynamoDbClientOptions.cs:8]: AWS ST takes care of this by giving the container the necessary AIM rights

File: src/RoadRegistry.BackOffice.Api/Program.cs [hunk 8]
```diff
@@ -118,6 +120,30 @@ public static IWebHostBuilder CreateWebHostBuilder(string[] args)
...
                         }
 
+                        var dynamoDbClientOptions = new DynamoDbClientOptions();
+                        hostContext.Configuration.GetSection(DynamoDbClientOptions.ConfigurationKey).Bind(dynamoDbClientOptions);
+
+                        if (dynamoDbClientOptions.ServerUrl is null)
+                        {
+                            builder
+                                .AddSingleton(new AmazonDynamoDBClient(
+                                    new BasicAWSCredentials(
```
[the remark this answers is not in the document]
Comment (lvermeulen) [reply src/RoadRegistry.BackOffice.Api/Program.cs:130]: specifically about passing the BasicAWSCredentials -- isn't necessary

**Review by YusufMavzer — APPROVED** [review]

**Pull request merged**